messaging/src/test/java/org/axonframework/eventstreaming/LegacyStreamableEventSourceTest.java

```diff
@@ -0,0 +1,474 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventstreaming;
+
+import org.axonframework.common.stream.BlockingStream;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.EventTestUtils;
+import org.axonframework.eventhandling.GenericTrackedEventMessage;
+import org.axonframework.eventhandling.GlobalSequenceTrackingToken;
+import org.axonframework.eventhandling.TrackedEventMessage;
+import org.axonframework.eventhandling.TrackingToken;
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.MessageStream;
+import org.axonframework.messaging.StreamableMessageSource;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.*;
+import org.mockito.*;
+import org.mockito.junit.jupiter.*;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
+/**
+ * Tests for the {@link LegacyStreamableEventSource} class, which it's an adapter for the deprecated
+ * {@link StreamableMessageSource}.
+ *
+ * @author Mateusz Nowak
+ * @since 5.0.0
+ */
+@ExtendWith(MockitoExtension.class)
+class LegacyStreamableEventSourceTest {
```

Comment:
Curious: I am wondering for how long we'd need this test class. Wouldn't we kill the `StreamableMessageSource`, and this have no need for the `LegacyStreamableEventSource`, pretty soon?

Reply:
Yes, if we both understand pretty soon by: adjusting the `TrackingEventProcessor` as well as all parts of the configuration :)

Reply:
Gotcha, thanks for the explanation, @MateuszNaKodach. Nothing wrong with being 100% certain things work as intended concerning the stream replacement. It's a rather important component, after all.
With your answer here, I am good concerning my curiosity. Hence, I'll resolve this comment.